third_party/nvfuser/csrc/ir_utils.cpp

```diff
@@ -798,5 +798,16 @@ std::string varName(const Val* val) {
   return name.str();
 }
 
+std::vector<int> fullToNoBroadcastMap(const std::vector<IterDomain*> ids) {
+  std::vector<int> full2nob_map(ids.size());
+  int no_broadcast_i = 0;
+  for (const auto i : c10::irange(ids.size())) {
+    if (!ids.at(i)->isBroadcast()) {
+      full2nob_map.at(i) = no_broadcast_i++;
```

Comment:
I agree that `unordered_map` provide better readability and error checking, but isn't `unordered_map` much slower compared to `vector`? I changed the code to initialize the vector with `std::numeric_limits<size_t>::max()`, so if an unexpected item is accessed, it will almost always lead to an out-of-bound error or a segfault.